activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -356,35 +356,32 @@ def insert_fixture(fixture, table_name)
       # Inserts a set of fixtures into the table. Overridden in adapters that require
       # something beyond a simple insert (eg. Oracle).
       def insert_fixtures(fixtures, table_name)
+        ActiveSupport::Deprecation.warn(<<-MSG.squish)
+          `insert_fixtures` is deprecated and will be removed in the next version of Rails.
+          Consider using `insert_fixtures_set` for performance improvement.
+        MSG
         return if fixtures.empty?
 
-        columns = schema_cache.columns_hash(table_name)
+        execute(build_fixture_sql(fixtures, table_name), "Fixtures Insert")
+      end
 
-        values = fixtures.map do |fixture|
-          fixture = fixture.stringify_keys
+      def insert_fixtures_set(fixture_set, tables_to_delete = [])
+        fixture_inserts = fixture_set.map do |table_name, fixtures|
+          next if fixtures.empty?
 
-          unknown_columns = fixture.keys - columns.keys
-          if unknown_columns.any?
-            raise Fixture::FixtureError, %(table "#{table_name}" has no columns named #{unknown_columns.map(&:inspect).join(', ')}.)
-          end
+          build_fixture_sql(fixtures, table_name)
+        end.compact
 
-          columns.map do |name, column|
-            if fixture.key?(name)
-              type = lookup_cast_type_from_column(column)
-              bind = Relation::QueryAttribute.new(name, fixture[name], type)
-              with_yaml_fallback(bind.value_for_database)
-            else
-              Arel.sql("DEFAULT")
+        total_sql = Array.wrap(combine_multi_statements(fixture_inserts, tables_to_delete))
```

Comment:
Sorry, that part was unclear:

I meant `insert_fixtures_set` can call `combine_multi_statements` a second time... or actually, even just combine them:

```ruby
table_deletes = tables_to_delete.map { |table| "DELETE FROM #{quote_table_name table}" }
total_sql = Array.wrap(combine_multi_statements(table_deletes + fixture_inserts))
```

Reply:
👍 it will make things even faster.  I was just unsure if people relied on the activerecord.sql notifications when deleting fixtures (https://github.com/rails/rails/pull/31422#discussion_r156528814).

Reply:
Yeah, I think it's fine for those to change; arguably people are more likely to be interested in the inserts than the deletes, and we're already radically changing those -- and no-one complained last time.

(If it's going to be doing deletes *and* inserts, I guess we should change the label on the `execute` to `"Fixtures Load"` or something, too.)